{
	"document_title": "CVE-2023-45805",
	"document_type": "SUSE CVE",
	"document_publisher": {
		"type": "Vendor",
		"contact_details": "security@suse.de",
		"issuing_authority": "SUSE Security Team"
	},
	"document_tracking": {
		"identification": {
			"id": "SUSE CVE-2023-45805"
		},
		"status": "Interim",
		"version": "1",
		"revision_history": {
			"revision": {
				"number": "2",
				"date": "2023-10-24T23:08:44Z",
				"description": "current"
			}
		},
		"initial_release_date": "2023-10-23T23:08:36Z",
		"current_release_date": "2023-10-24T23:08:44Z",
		"generator": {
			"engine": "cve-database/bin/generate-cvrf-cve.pl",
			"date": "2020-12-27T01:00:00Z"
		}
	},
	"document_notes": [
		{
			"text": "CVE-2023-45805",
			"title": "CVE",
			"type": "Summary"
		},
		{
			"text": "pdm is a Python package and dependency manager supporting the latest PEP standards. It's possible to craft a malicious `pdm.lock` file that could allow e.g. an insider or a malicious open source project to appear to depend on a trusted PyPI project, but actually install another project. A project `foo` can be targeted by creating the project `foo-2` and uploading the file `foo-2-2.tar.gz` to pypi.org. PyPI will see this as project `foo-2` version `2`, while PDM will see this as project `foo` version `2-2`. The version must only be `parseable as a version` and the filename must be a prefix of the project name, but it's not verified to match the version being installed. Version `2-2` is also not a valid normalized version per PEP 440. Matching the project name exactly (not just prefix) would fix the issue. When installing dependencies with PDM, what's actually installed could differ from what's listed in `pyproject.toml` (including arbitrary code execution on install). It could also be used for downgrade attacks by only changing the version. This issue has been addressed in commit `6853e2642df` which is included in release version `2.9.4`. Users are advised to upgrade. There are no known workarounds for this vulnerability.",
			"title": "Mitre CVE Description",
			"type": "Description"
		},
		{
			"text": "The CVRF data is provided by SUSE under the Creative Commons License 4.0 with Attribution (CC-BY-4.0).",
			"title": "Terms of Use",
			"type": "Legal Disclaimer"
		}
	],
	"document_references": [
		{
			"type": "Self",
			"url": "https://www.suse.com/support/security/rating/",
			"description": "SUSE Security Ratings"
		}
	],
	"product_tree": {
		"branch": [
			{
				"type": "Product Family",
				"name": "openSUSE Tumbleweed",
				"branch": [
					{
						"type": "Product Name",
						"name": "openSUSE Tumbleweed",
						"full_product_name": {
							"text": "openSUSE Tumbleweed",
							"product_id": "openSUSE Tumbleweed",
							"cpe": "cpe:/o:opensuse:tumbleweed"
						}
					}
				],
				"full_product_name": {}
			},
			{
				"type": "Product Version",
				"name": "python310-unearth-0.12.0-1.1",
				"full_product_name": {
					"text": "python310-unearth-0.12.0-1.1",
					"product_id": "python310-unearth-0.12.0-1.1"
				}
			},
			{
				"type": "Product Version",
				"name": "python311-unearth-0.12.0-1.1",
				"full_product_name": {
					"text": "python311-unearth-0.12.0-1.1",
					"product_id": "python311-unearth-0.12.0-1.1"
				}
			},
			{
				"type": "Product Version",
				"name": "python39-unearth-0.12.0-1.1",
				"full_product_name": {
					"text": "python39-unearth-0.12.0-1.1",
					"product_id": "python39-unearth-0.12.0-1.1"
				}
			}
		],
		"relationship": [
			{
				"product_reference": "python310-unearth-0.12.0-1.1",
				"relation_type": "Default Component Of",
				"relates_to_product_reference": "openSUSE Tumbleweed"
			},
			{
				"product_reference": "python311-unearth-0.12.0-1.1",
				"relation_type": "Default Component Of",
				"relates_to_product_reference": "openSUSE Tumbleweed"
			},
			{
				"product_reference": "python39-unearth-0.12.0-1.1",
				"relation_type": "Default Component Of",
				"relates_to_product_reference": "openSUSE Tumbleweed"
			}
		]
	},
	"vulnerability": [
		{
			"notes": [
				{
					"text": "pdm is a Python package and dependency manager supporting the latest PEP standards. It's possible to craft a malicious `pdm.lock` file that could allow e.g. an insider or a malicious open source project to appear to depend on a trusted PyPI project, but actually install another project. A project `foo` can be targeted by creating the project `foo-2` and uploading the file `foo-2-2.tar.gz` to pypi.org. PyPI will see this as project `foo-2` version `2`, while PDM will see this as project `foo` version `2-2`. The version must only be `parseable as a version` and the filename must be a prefix of the project name, but it's not verified to match the version being installed. Version `2-2` is also not a valid normalized version per PEP 440. Matching the project name exactly (not just prefix) would fix the issue. When installing dependencies with PDM, what's actually installed could differ from what's listed in `pyproject.toml` (including arbitrary code execution on install). It could also be used for downgrade attacks by only changing the version. This issue has been addressed in commit `6853e2642df` which is included in release version `2.9.4`. Users are advised to upgrade. There are no known workarounds for this vulnerability.",
					"title": "Vulnerability Description",
					"type": "General"
				}
			],
			"cve": "CVE-2023-45805",
			"product_statuses": [
				{
					"type": "Fixed",
					"product_id": [
						"openSUSE Tumbleweed:python310-unearth-0.12.0-1.1",
						"openSUSE Tumbleweed:python311-unearth-0.12.0-1.1",
						"openSUSE Tumbleweed:python39-unearth-0.12.0-1.1"
					]
				}
			],
			"threats": [
				{
					"type": "Impact",
					"description": "important"
				}
			],
			"cvss_score_sets": {
				"score_set_v_2": {},
				"score_set_v_3": {
					"base_score_v_3": "7.8",
					"vector_v_3": "CVSS:3.1/AV:L/AC:L/PR:N/UI:R/S:U/C:H/I:H/A:H"
				}
			}
		}
	]
}
